src/core/data/ndarray/ndarray_vector.h

```diff
@@ -5,12 +5,216 @@
+
+            this->data_ = std::move(source.data_);
+            return *this;
+        }
+
+
+        template<typename... Indexes>
+        DataType& operator()(Indexes... indexes)
+        {
+            auto params = std::tuple<Indexes...>{indexes...};
+            static_assert(sizeof...(Indexes) == dim);
+
+            if constexpr (dim == 1)
+            {
+                auto i = std::get<0>(params);
+
+                static_assert(std::is_unsigned<decltype(i)>::value);
+
+                return this->data_[i];
+            }
+
+            if constexpr (dim == 2)
+            {
+                auto i = std::get<0>(params);
+                auto j = std::get<1>(params);
+
+                static_assert(std::is_unsigned<decltype(i)>::value);
+                static_assert(std::is_unsigned<decltype(j)>::value);
+
+                return this->data_[j + i * nCells_[1]];
+            }
+
+            if constexpr (dim == 3)
+            {
+                auto i = std::get<0>(params);
+                auto j = std::get<1>(params);
+                auto k = std::get<2>(params);
+
+                static_assert(std::is_unsigned<decltype(i)>::value);
+                static_assert(std::is_unsigned<decltype(j)>::value);
+                static_assert(std::is_unsigned<decltype(k)>::value);
+
+                return this->data_[k + j * nCells_[2] + i * nCells_[1] * nCells_[2]];
+            }
+
+            if constexpr (dim != 1 && dim != 2 && dim != 3)
+            {
+                return 0;
+            }
+        }
+
+        template<typename... Indexes>
+        DataType const& operator()(Indexes... indexes) const
+        {
+            auto params = std::tuple<Indexes...>{indexes...};
+            static_assert(sizeof...(Indexes) == dim);
+
+            if constexpr (dim == 1)
+            {
+                auto i = std::get<0>(params);
+
+                static_assert(std::is_unsigned<decltype(i)>::value);
+
+                return this->data_[i];
+            }
+
+            if constexpr (dim == 2)
+            {
+                auto i = std::get<0>(params);
+                auto j = std::get<1>(params);
+
+                static_assert(std::is_unsigned<decltype(i)>::value);
+                static_assert(std::is_unsigned<decltype(j)>::value);
+
+                return this->data_[j + i * nCells_[1]];
+            }
+
+            if constexpr (dim == 3)
+            {
+                auto i = std::get<0>(params);
+                auto j = std::get<1>(params);
+                auto k = std::get<2>(params);
+
+                static_assert(std::is_unsigned<decltype(i)>::value);
+                static_assert(std::is_unsigned<decltype(j)>::value);
+                static_assert(std::is_unsigned<decltype(k)>::value);
+
+                return this->data_[k + j * nCells_[2] + i * nCells_[1] * nCells_[2]];
+            }
+
+            if constexpr (dim != 1 && dim != 2 && dim != 3)
+            {
+                return 0;
```

Comment:
I don't think we want to return 0 (or anything for that matter) in that case.. if dim is invalid we should not even be here in the first place.

Reply:
now, it throws an runtime error,